mlflow/pipelines/steps/train.py

```diff
@@ -841,14 +882,6 @@ def _safe_dump_with_numeric_values(self, data, file, **kwargs):
                 processed_data[key] = int(value)
             else:
                 processed_data[key] = value
-        return yaml.safe_dump(processed_data, file, **kwargs)
 
-    def _is_tuning_param_equal(self, tuning_param, logged_param):
-        if isinstance(tuning_param, int):
-            return tuning_param == int(logged_param)
-        elif isinstance(tuning_param, float):
-            return tuning_param == float(logged_param)
-        elif isinstance(tuning_param, str):
-            return tuning_param.strip() == logged_param.strip()
-        else:
-            return tuning_param == logged_param
+        if len(processed_data) > 0:
```

Comment:
Why is the length check necessary?

Reply:
this avoids writing empty curly braces into the yaml. otherwise, our yaml will look like
```
# hardcoded
{}
# tuned
alpha: 0.5
```

which cannot be read
```
>>> with open('tmp.yaml', 'r') as f:
...     x = yaml.safe_load(f)
...     print(x)
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "/Users/prithvi.kannan/Library/Python/3.8/lib/python/site-packages/yaml/__init__.py", line 125, in safe_load
    return load(stream, SafeLoader)
  File "/Users/prithvi.kannan/Library/Python/3.8/lib/python/site-packages/yaml/__init__.py", line 81, in load
    return loader.get_single_data()
  File "/Users/prithvi.kannan/Library/Python/3.8/lib/python/site-packages/yaml/constructor.py", line 49, in get_single_data
    node = self.get_single_node()
  File "/Users/prithvi.kannan/Library/Python/3.8/lib/python/site-packages/yaml/composer.py", line 39, in get_single_node
    if not self.check_event(StreamEndEvent):
  File "/Users/prithvi.kannan/Library/Python/3.8/lib/python/site-packages/yaml/parser.py", line 98, in check_event
    self.current_event = self.state()
  File "/Users/prithvi.kannan/Library/Python/3.8/lib/python/site-packages/yaml/parser.py", line 171, in parse_document_start
    raise ParserError(None, None,
yaml.parser.ParserError: expected '<document start>', but found '<block mapping start>'
  in "tmp.yaml", line 4, column 1
```

Reply:
Great!